### ISSUE TITLE
Keyboard navigation in dropdown menus that include separators is broken

### ISSUE THREAD

**Issue by microbit-robert** [RaspberryPiFoundation/blockly#9473]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

If a field_dropdown menu contains a separator, keyboard navigation doesn't work properly when moving up and down through the menu. This is due to a mix of methods being used in `highlightHelper` and `highlightPrevious` / `highlightNext`. `this.getMenuItems()` filters out separators, but `this.menuItems` does not and they are used interchangeably. 

### Reproduction steps

1. 
2. 
3.

### Stack trace

```shell

```

**Comment by BenHenning** [comment]
FYI This will need to get pulled into the screen reader branch via a separate merge PR.

Edit: Actually will reopen this so that we don't forget to do that (since we rarely have need to merge in the latest `main` changes into the experimental branch).
